internal/config/types.go

```diff
@@ -36,3 +59,36 @@ type Config struct {
 	Metrics                 Metrics       `yaml:"metrics" json:"metrics" env-prefix:"KOKO_METRICS_"`
 	DisableAnonymousReports bool          `yaml:"disable_anonymous_reports" json:"disable_anonymous_reports" env-prefix:"KOKO_DISABLE_ANONYMOUS_REPORTS"` //nolint:lll
 }
+
+// fetchFileContents gathers file contents from the provided fields and sets it on the applicable string field.
+func (t *TLS) fetchFileContents() error {
+	type fieldData struct {
+		name     string
+		filename string
+		ptr      *string
+	}
+
+	for _, d := range []fieldData{
+		{"root CA", t.RootCAFile, &t.RootCA},
+		{"certificate", t.CertificateFile, &t.Certificate},
+		{"key", t.KeyFile, &t.Key},
+	} {
+		// No filename set for the field, so we can ignore.
+		if d.filename == "" {
+			continue
+		}
+
+		if *d.ptr != "" {
+			return fmt.Errorf("cannot set both %s file & raw string", d.name)
```

Comment:
Would it be useful to also print the filename here and in the next error?

Reply:
For this error, maybe. The error gets wrapped into the "unable to set MySQL DB options" error, and I was thinking that's more than enough as it refers to the config. No opposition to adding the file name here though.

As for the next error, nah, the Golang open file tooling produces an error with the file path:
```
unable to set MySQL DB options: unable to read root CA file contents: open /this/file/path/does/not/exist.pem: no such file or directory
```